src/main/java/edu/harvard/iq/dataverse/DatasetFieldServiceBean.java

```diff
@@ -561,6 +562,10 @@ public void process(HttpResponse response, HttpContext context) throws HttpExcep
 
     private String replaceRetrievalUriParam(String retrievalUri, String paramName, String value) {
 
+        if(StringUtils.isBlank(paramName) || StringUtils.isBlank(value)) {
+            return retrievalUri;
+        }
+
```

Comment:
It seems like line 501 above is the only thing that could cause the NPE? And in that case the paramName can't be null but the value could be? I wonder if checking on line 571 to see if retrievalUri.contains(paramName) would be a better check - if the param name is in the Uri and it isn't replaced, that's a problem you wouldn't want to silently ignore.

Reply:
CVOC is sooo complexe...

- you are right the only case where there is NPE can happen is : L501 `retrievalUri = replaceRetrievalUriParam(retrievalUri, f.getDatasetFieldType().getName(), f.getValue());` when value is null which can happen with termUri filled but when a parameter set in retrievalUri has no value.
- _retrievalUri.contains(paramName) would be a better check - if the param name is in the Uri and it isn't replaced, that's a problem you wouldn't want to silently ignore_ if param is null, empty or wrong I just do nothing, either it cannot be done, data are wrong or json setting is wrong. But that being said, a "if paramName does not exist in retrievalUri then logger.warning" can be done.
- _the code shouldn't just call the remote service with a broken URL, so maybe the NPE should be caught and used to avoid the call to the service instead?_ hum, you are right, for a wrong parameter we could simply `return;` instead of performing the call.

Reply:
@qqmyers  79b8d89 modifications have been made, if you can review / approve :)